cypress/e2e/one-to-one-chat/one-to-one-chat.spec.ts

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import { SynapseInstance } from "../../plugins/synapsedocker";
+import type { Credentials } from "../../support/synapse";
+
+describe("1:1 chat room", () => {
+    let synapse: SynapseInstance;
+    let user2: Credentials;
+
+    const username = "user1234";
+    const password = "p4s5W0rD";
+
+    beforeEach(() => {
+        cy.startSynapse("default").then((data) => {
+            synapse = data;
+
+            cy.initTestUser(synapse, "Jeff");
+            cy.registerUser(synapse, username, password).then((credentials) => {
+                user2 = credentials;
+                cy.visit(`/#/user/${user2.userId}?action=chat`);
+            });
+        });
+    });
+
+    afterEach(() => {
+        cy.stopSynapse(synapse);
+    });
+
+    it("should open new 1:1 chat room after leaving the old one", () => {
+        // leave 1:1 chat room
+        cy.contains(".mx_RoomHeader_nametext", username).click();
+        cy.contains('[role="menuitem"]', "Leave").click();
+        cy.get('[data-testid="dialog-primary-button"]').click();
+
+        // wait till the room was left
+        cy.get('[role="group"][aria-label="Historical"]').within(() => {
+            cy.contains(".mx_RoomTile", username);
+        });
+
+        // open new 1:1 chat room
+        cy.visit(`/#/user/${user2.userId}?action=chat`);
+        cy.contains(".mx_RoomHeader_nametext", username);
```

Comment:
Wondering why we consider this assertion to be enough?
I've just checked, and leaving a room does not change its name, it only moves it to the `Historical` section.

However this assertion checks the room name.

Reply:
For me, the old room is not displayed correctly anymore and doesn't show a header where the name would be:

<img width="1013" alt="image" src="https://user-images.githubusercontent.com/720821/211532595-4a6321a2-75bf-4f82-a389-911d33ce2a9e.png">

(I can't click "join the discussion" because I am no longer in the room and can't join without being re-invited. Sure the other user then has two rooms, but that's how it is)

So I would assume that the assertion could only be fulfilled when I am in a new 1:1 room that was properly created.